### ISSUE TITLE
REST API: get total and current duration, iterations, etc.

### ISSUE THREAD

**Issue by xshrim** [grafana/k6#1111]
How can i get the current and total duration/iterations of a load testing from the rest api?

**Comment by na--** [comment]
You currently can't get them directly, sorry, though with some simple workarounds, you might be able to get what you need. You can get the total duration/iteration/etc. options by returning them from `setup()` like this:
```js
import http from "k6/http";
import { sleep } from "k6";

export let options = {};

export function setup() {
    return {
        myOptions: {
            vus: options.vus,
            duration: options.duration,
            // ...
        }
    }
}

export default function () {
    http.get(`https://test.loadimpact.com/`);
    sleep(1);
}
```

This works because the consolidated k6 options (CLI flags + env vars + JS options + JSON options) will be propagated back to the JS `options` object before the `setup()` or VU execution. So, running that script with `k6 run --vus 5 --duration 20m` and then running `curl http://localhost:6565/v1/setup` will get you something like:
```json
{"data":{"type":"setupData","id":"default","attributes":{"data":{"myOptions":{"duration":"20m0s","vus":5}}}}}
```

And running `curl http://localhost:6565/v1/metrics/iterations` (to get the value of the `iterations` metric) will get you something like:
```
{"data":{"type":"metrics","id":"iterations","attributes":{"type":"counter","contains":"default","tainted":null,"sample":{"count":19,"rate":3.457626896665038}}}}
```

You can get the current VUs (useful for when you have ramping up or down of VUs via `stages`) via `curl http://localhost:6565/v1/status` or `curl http://localhost:6565/v1/metrics/vus`. However, there isn't a metric for the current elapsed test duration, and while you can probably create a [custom one](https://docs.k6.io/docs/result-metrics#section-custom-metrics) that you update in the script, you might be better off with just an external timer. 

I'll leave this issue open, since it makes sense to have these directly and in a single place the API, but because of the somewhat easy workarounds, it's definitely going to be a low priority one. And it's definitely going to be after #1007 is merged, because otherwise we'd have to do it twice...

**Comment by mstoykov** [comment]
I am closing this due to lack of interest. 

We are very unlikely to put this in v1 and v2 likely will have similar functionality as it will need to supported distributed execution.

see https://github.com/grafana/k6/issues/3497
